### ISSUE TITLE
Indentation and synchronized block

### ISSUE THREAD

**Issue by vitkud** [checkstyle/checkstyle#580]
I think there is an issue with indentation tracking:
```java
public final class Singleton {
    // ...
    public static Singleton getInstance() {
        if (instance == null)
            synchronized (Singleton.class) {
                if (instance == null) // 'if' have incorrect indentation level 16, expected level should be 12.
                    instance = new Singleton();
            }
        return instance;
    }
}
```
Here is my checkstyle configuration:
```xml
<module name="Checker">
  <module name="TreeWalker">
    <module name="Indentation">
    </module>
  </module>
</module>

```

**Comment by liscju** [comment]
Problem is that for line with 'if (instance == null)' its parent( '{' in line with synchronized) is asked for suggested child level in SlistHandler.suggestedChildLevel. To check what level should suggest , it checks for parent ExpressionHandler - in this situation IfHandler(If in line with 'instance == null') because there is not Handler for Synchronized statements. In suggestedChildLevel it checks whether 
getParent() instanceof BlockParentHandler && !(getParent() instanceof SlistHandler) 
in this case true so it ask IfHandler(in line with 'instance == null') to suggest child level. It suggest same intendation as for synchronized element. If we put '{' after if (in line with 'instance == null') it would make getParent() returns that '{' what would make 
getParent() instanceof BlockParentHandler && !(getParent() instanceof SlistHandler) 
false and suggestion for child level in '{' (in line with synchronized) would be make according to this '{' not its parent. This behaviour has in my view subtle assumption that if getParent() is different from SListHandler this SListHandler is a 'parantheses' for this handler(what is not a case in this situation because there is not synchronize handler and If is without '{' ). To correct this behaviour it can be introduced new SynchronizeHandler to handle this. I hope im understandable, and if maxvetrenko would be busy or something i can take this task and fix this.

**Comment by romani** [comment]
@liscju, please do some formatting for your comment.

**Comment by romani** [comment]
merged.
